net/client_grpc.go

```diff
@@ -157,3 +160,39 @@ func (p *proxyClient) Private(c context.Context, in *drand.PrivateRandRequest, o
 func (p *proxyClient) DistKey(c context.Context, in *drand.DistKeyRequest, opts ...grpc.CallOption) (*drand.DistKeyResponse, error) {
 	return p.s.DistKey(c, in)
 }
+
+//ControlClient is a struct that implement control.ControlClient and is used to request
+//a Share to a ControlListener on a specific port
+type ControlClient struct {
+	conn   *grpc.ClientConn
+	client control.ControlClient
+}
+
+// NewControlClient creates a client connection to the given target (localhost:8888)
+func NewControlClient(port string) ControlClient {
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial(fmt.Sprintf("%s:%s", "localhost", port), grpc.WithInsecure())
+	if err != nil {
+		slog.Fatalf("control: did not connect: %s", err)
+		return ControlClient{}
+	}
+	c := control.NewControlClient(conn)
+	return ControlClient{conn: conn, client: c}
+}
+
+// Share requestsa nd returns the private share
+func (c ControlClient) Share() (map[string]string, error) {
```

Comment:
Why is this returning a map ? Would it not make more sense if it was just returning the share, as a `share.Share` struct ?